pkg/runctl/controller.go

```diff
@@ -428,25 +405,15 @@ func (c *Controller) syncHandler(key string) error {
 			if serrors.IsRecoverable(err) {
 				return err
 			}
-			if errClean := c.changeState(pipelineRun, api.StateCleaning); errClean != nil {
-				return errClean
-			}
 			pipelineRun.StoreErrorAsMessage(err, "running failed")
-			return c.commitStatusAndMeter(pipelineRun)
+			// TODO: why don't we set a result here? Maybe prepareCleaningAndCountResult ...
```

Comment:
IMHO there should be an ErrorInfra set here.

Reply:
done